src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +190,19 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        if action_type and action_type == "variable":
+            form_var = FormVariable.objects.get(key=variable)
+
+            if form_var.data_type == "date":
+                try:
+                    date.fromisoformat(data.get("action").get("value"))
+                except ValueError:
+                    raise serializers.ValidationError(
+                        {
+                            "action.value": "Value for date variable must be in isoformart (yyyy-mm-dd)"
```

Comment:
```suggestion
                            "action.value": "Value for date variable must be in isoformat (yyyy-mm-dd)"
```

Reply:
Although maybe this could be simplified to `Value for date variable must be in format yyyy-mm-dd? :thinking: